syntax = "proto3";

package payment;

// Service to demo gRPC in processing payments
service PaymentService {
    rpc ProcessPayment (PaymentRequest) returns (PaymentResponse);
}

message PaymentRequest{
    string card_number = 1;
    double amount = 2;
    string currency =3;
}

message PaymentResponse{
    bool success = 1;
    string transaction_id = 2;
    string message = 3;
    string server_note = 4; 
}